openff/nagl_models/_dynamic_fetch.py

```diff
@@ -0,0 +1,42 @@
+import urllib.request
+import json
+from packaging.version import Version
+import platformdirs
+import pathlib
+
+
+RELEASES_URL = "https://api.github.com/repos/openforcefield/openff-nagl-models/releases"
+
+release_metadata: list[dict] = json.load(urllib.request.urlopen(RELEASES_URL))
+
+# tags with "v" prefix can't easily be sorted, but the result of passing through Version
+# are not necessarily 1:1 with the metadata in the releases, keep both and map between
+releases: dict[Version:str] = {
+    Version(release["tag_name"]): release for release in release_metadata
+}
+
+
+def get_most_recent_asset_url(filename: str) -> str:
+    pathlib.Path(platformdirs.user_cache_path() / "OPENFF_NAGL_MODELS").mkdir(
+        exist_ok=True
+    )
+
+    cached_path = platformdirs.user_cache_path() / "OPENFF_NAGL_MODELS" / filename
+
+    if cached_path.exists():
+        return cached_path.as_posix()
+
+    for version in reversed(sorted(releases)):
```

Comment:
I'm pretty sure this would still work after #43 (equivalently: I think PEP 440 is compatible with CalVer)

```python
>>> from packaging.version import Version
>>> [
...     *reversed(
...         sorted(
...             [
...                 Version(tag)
...                 for tag in ["2024.10.0", "2024.10.1", "2025.1.0", "2025.12.0"]
...             ]
...         )
...     )
... ]
[<Version('2025.12.0')>, <Version('2025.1.0')>, <Version('2024.10.1')>, <Version('2024.10.0')>]
>>>